app.rb

```diff
@@ -0,0 +1,71 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+  end
+
+  def list_all_books
+    # Use the Book class method to get all books
+    Book.all.each do |book|
+      puts "Title: #{book.title} | Author: #{book.author} ID: #{book.id}"
+    end
+  end
+
+  def list_all_people
+    # Use the Person class method to get all people
+    Person.all.each do |person|
+      puts "Name: #{person.correct_name} | Age: #{person.age} | ID: #{person.id}"
+    end
+  end
+
+  def create_person(name, age, type, options = {})
+    case type
+    when 'student'
+      # Create a new student object with the given attributes
+      student = Student.new(age, options[:classroom], name: name, parent_permission: options[:parent_permission])
+      # Add the student to the people array
+      @people << student
+      # Print a success message
+      puts "Student #{student.name} created successfully!"
+
+    when 'teacher'
+      # Create a new teacher object with the given attributes
+      teacher = Teacher.new(age, options[:specialization], name: name, parent_permission: options[:parent_permission])
+      # Add the teacher to the people array
```

Comment:
- [ ] Please your app should work exactly the same as in the [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) and I think if a person is a teacher you don't have to ask for `parent_permission` as you can see in the [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) but your app is still asking for `parent_permission` if even the person is a teacher so make sure that you fix that. :+1: